lib/katello/tasks/update_content_default_http_proxy.rake

```diff
@@ -0,0 +1,57 @@
+require File.expand_path("../engine", File.dirname(__FILE__))
+
+namespace :katello do
+  desc "Sets the content default http proxy to an existing http proxy based on supplied URL."
+  task :update_default_http_proxy => :environment do
+    setting = ::Setting::Content.where(name: 'content_default_http_proxy').first
+    options = {}
+    o = OptionParser.new
+    o.banner = "Usage: rake katello:update_content_default_http_proxy [options]"
```

Comment:
i've not ever written a rake task to use an option parser, how do you execute it with options?

```
$ rake  katello:update_default_http_proxy 
ERROR: Missing required option for --name HTTP_PROXY_NAME

[vagrant@devel2 foreman]$ rake  katello:update_default_http_proxy  --name foo
invalid option: --name
[vagrant@devel2 foreman]$ rake  katello:update_default_http_proxy  --name=foo
invalid option: --name=foo
[vagrant@devel2 foreman]$ rake  --name=foo katello:update_default_http_proxy  
invalid option: --name=foo
[vagrant@devel2 foreman]$ rake  --name foo katello:update_default_http_proxy  
invalid option: --name
[vagrant@devel2 foreman]$ rake  katello:update_default_http_proxy  -n foo
/home/vagrant/git/foreman/lib/foreman.rb:8: warning: already initialized constant Foreman::UUID_REGEXP
/home/vagrant/git/foreman/lib/foreman.rb:8: warning: previous definition of UUID_REGEXP was here
** Invoke katello:update_default_http_proxy (first_time)
** Invoke environment (first_time)
** Execute (dry run) environment
** Execute (dry run) katello:update_default_http_proxy
rake aborted!
Don't know how to build task 'foo' (See the list of available tasks with `rake --tasks`)
/home/vagrant/.gem/ruby/gems/rake-12.3.2/exe/rake:27:in `<top (required)>'
(See full trace by running task with --trace)

```

Reply:
Take a look at the PR comment near the top for how I used it.

`bundle exec rails katello:update_default_http_proxy -- -n another_proxy -u http://admin:admin@192.168.23.131:8888`

Note that you have to use the `--` so that the options are seen by rake, I think otherwise they are interpreted as options for the ruby interpreter.

Reply:
ohh right the -- .  Mind updating the usage to include that?

Reply:
oh and you might print usage if name is missing, otherwise you have to know the -- trick to even print help text

Reply:
It was frustrating to figure out because the recommended pattern for using optparse is to supply args in the task definition. However, that approach had issues when trying to run them via a test. So I decided to use the ARGV array instead.

Reply:
yeah, i think what you have is fine, its probably better than just using ENV variables which we typically do, we just want to make sure its obvious how to run it when you don't pass any options :)